internal/backend/auth/authhttpmiddleware/authmiddleware.go

```diff
@@ -83,6 +83,11 @@ func newSessionsMiddleware(next http.Handler, sessions authsessions.Store) http.
 				return
 			}
 
+			if session == nil {
```

Comment:
why is this needed? if we error out here it wont get to the tokens middleware.

Reply:
we have a problematic case

both session and tokens are optional

```go
	Sessions authsessions.Store `optional:"true"`
	Tokens   authtokens.Tokens  `optional:"true"`
```

so if tokens are optional, sessions has to validate or fail if session is invalid

it also makes an order dependency which is not explicit

we should refactor both checks to one function and make it super explicit and readable, no need for optional checks

```go
if session == nil && token == nil {
   return 401
}
```

what do you want to do ? fix it on this PR ? merge it and open another PR ?

Reply:
my bad, it should work, I'll recheck required config

Reply:
seems legit on  Itay's debug dashboard as well